shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,154 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.dto.ProductCombinedDto;
+import ro.msg.learning.shop.dto.SupplierDto;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.model.ProductCategory;
+import ro.msg.learning.shop.model.Supplier;
+import ro.msg.learning.shop.repository.IProductCategoryInterfaceRepository;
+import ro.msg.learning.shop.repository.IProductInterfaceRepository;
+import ro.msg.learning.shop.repository.ISupplierInterfaceRepository;
+import ro.msg.learning.shop.service.exceptions.NotFoundException;
+import ro.msg.learning.shop.service.exceptions.ProductException;
+import ro.msg.learning.shop.utils.Mapper;
+import ro.msg.learning.shop.utils.Status;
+
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class ProductService {
+    @Autowired
+    private IProductInterfaceRepository productInterfaceRepository;
+    @Autowired
+    private ISupplierInterfaceRepository supplierInterfaceRepository;
+    @Autowired
+    private IProductCategoryInterfaceRepository productCategoryInterfaceRepository;
+
+
+
+
+    private Mapper mapper = new Mapper();
+
+    public ProductService(IProductInterfaceRepository productInterfaceRepository,ISupplierInterfaceRepository supplierInterfaceRepository,IProductCategoryInterfaceRepository productCategoryInterfaceRepository)
+    {
+        this.productInterfaceRepository=productInterfaceRepository;
+        this.productCategoryInterfaceRepository= productCategoryInterfaceRepository;
+        this.supplierInterfaceRepository=supplierInterfaceRepository;
+    }
+
+
+    public ProductCategory saveProductCategory(ProductCategory productCategory)
```

Comment:
Please follow the Single Responsability Principle (create a service for each object: Product, ProductCategory, Supplier)